nifi-nar-bundles/nifi-html-bundle/nifi-html-processors/src/main/java/org/apache/nifi/GetHTMLElement.java

```diff
@@ -0,0 +1,243 @@
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.io.StreamCallback;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Collections;
+
+@Tags({"get", "html", "dom", "css", "element"})
+@CapabilityDescription("Parses HTML input using CSS selector syntax and creates a new flowfile containing the extracted" +
+        " element content for each matching CSS selector.")
+@SeeAlso({ModifyHTMLElement.class, PutHTMLElement.class})
+@WritesAttributes({@WritesAttribute(attribute="HTMLElement", description="Flowfile attribute where the element result" +
+        " parsed from the HTML using the CSS selector syntax are placed if the destination is a flowfile attribute.")})
+public class GetHTMLElement
+        extends AbstractHTMLProcessor {
+
+    public static final String HTML_ELEMENT_ATTRIBUTE_NAME = "HTMLElement";
+    public static final String DESTINATION_ATTRIBUTE = "flowfile-attribute";
+    public static final String DESTINATION_CONTENT = "flowfile-content";
+
+    public static final PropertyDescriptor PREPEND_ELEMENT_VALUE = new PropertyDescriptor
+            .Builder().name("Prepend Element value")
+            .description("Prepends the specified value to the resulting Element")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor APPEND_ELEMENT_VALUE = new PropertyDescriptor
+            .Builder().name("Append Element value")
+            .description("Appends the specified value to the resulting Element")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+    public static final PropertyDescriptor ATTRIBUTE_KEY = new PropertyDescriptor
+            .Builder().name("Attribute Name")
+            .description(("When getting the value of an element attribute this value is used as the key to determine" +
+                    " which attribute on the selected element should be retrieved."))
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+
+
+    public static final PropertyDescriptor OUTPUT_TYPE = new PropertyDescriptor.Builder()
+            .name("Output Type")
+            .description("Controls the type of value that is retrieved from the element. " +
+                    ELEMENT_HTML + "," + ELEMENT_TEXT + ", " + ELEMENT_ATTRIBUTE + " or " + ELEMENT_DATA)
```

Comment:
We generally like to avoid listing the Allowable Values in the description. The UI will automatically provide this information based on the Allowable Values given; this also allows each of the Allowable Values to provide descriptions, if appropriate.